### PR TITLE
[Storage] Added unc to share property for mounting

### PR DESCRIPTION
**Pull request by williexu**
---
Fixes: https://github.com/Azure/azure-cli/issues/1589

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by troydai** [comment]
Not really sure about this. Originally we return whatever the REST API response body and this change essentially add an out-of-spec property to it. Personally, this is not a good idea.

**Comment by williexu** [comment]
After discussion, I believe the best way to incorporate this property is through a new command, similar to the other `url` commands in storage. Will make this change.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -435,6 +435,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('storage share') as c:
         c.argument('share_name', share_name_type, options_list=('--name', '-n'))
 
+    with self.argument_context('storage share url') as c:
+        c.argument('unc', action='store_true', help='Output UNC network path.')
+        c.argument('protocol', arg_type=get_enum_type(['http', 'https'], 'https'), help='Protocol to use.')
```
Comment (adewaleo) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: Will adding the parameter name in the call to get_enum_type e.g. "default='https'" make things clearer?

Comment (adewaleo) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: Also do we want to unittest / scenario test protocol?

File: src/command_modules/azure-cli-storage/HISTORY.rst [hunk 2]
```diff
@@ -6,6 +6,7 @@ Release History
 2.2.3
 +++++
 * Fix `az storage cors list` output formatting, all items show correct "Service" key
+* Add "unc" property to `storage share show` output to be used for mounting the share.
```
[the remark this answers is not in the document]
Comment (williexu) [reply src/command_modules/azure-cli-storage/HISTORY.rst]: https://whatis.techtarget.com/definition/Universal-Naming-Convention-UNC

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py [hunk 3]
```diff
@@ -438,6 +438,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('storage share list') as c:
         c.ignore('marker')  # https://github.com/Azure/azure-cli/issues/3745
 
+    with self.argument_context('storage share show') as c:
+        c.argument('snapshot', help='A string that represents the snapshot version, if applicable.')
```
[the remark this answers is not in the document]
Comment (williexu) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: Yep, I've changed the implementation so this no longer applies.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py (hunk 1 above)
Comment (williexu) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: Ideally, yes, however, it's infeasible to test every parameter for every command in larger modules like storage. 
As the `url` command makes no service calls and `protocol` is simply injected into a string, I don't think it is as useful to test.

**Pull request merged**